File: src/main/java/klaxon/akka/examples/java/helloworld/HelloWorld.java
package klaxon.akka.examples.java.helloworld;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

/**
 * <p>date 9/16/14 </p>
 * @author klaxon
 */
public class HelloWorld {

    public static void main(String[] args) {
        ActorSystem actorSystem = ActorSystem.create("helloWorld");
        ActorRef actorRef = actorSystem.actorOf(Props.create(GreetingActor.class));

        actorRef.tell(new Greeting("World"), ActorRef.noSender());
        actorRef.tell(new Greeting("Sun"), ActorRef.noSender());
        actorRef.tell(new Greeting("Moon"), ActorRef.noSender());

        actorSystem.shutdown();
    }

}

File: src/main/scala/klaxon/akka/examples/scala/HelloWorld.scala
package klaxon.akka.examples.scala

import akka.actor.{Actor, ActorLogging, ActorSystem, Props}

/**
 * <p>date 9/16/14 </p>
 * @author klaxon
 */
object HelloWorld {

  def main(args: Array[String]) {
    val actorSystem = ActorSystem("helloWorld")
    val greeter = actorSystem.actorOf(Props[GreetingActor], "greeter")

    greeter ! Greeting("World")
    greeter ! Greeting("Sun")
    greeter ! Greeting("Moon")

    actorSystem.shutdown()
  }

}

case class Greeting(who: String)

class GreetingActor extends Actor with ActorLogging {

  override def receive: Receive = {
    case Greeting(who) => log.info("Hello " + who)
  }
}
